README.md

```diff
@@ -65,67 +105,173 @@ assert new_obj == obj
+        ...
 
     @classmethod
-    def _dump_content(cls, data: Any) -> bytes:
-        return serialize_foo(data)  # custom serialization logic
-```
+    def _read_content(cls, path: str) -> bytes:
+        # Custom logic for reading bytes from a path using "foo" protocol
+        ...
+
+    @classmethod
+    def _create_content(cls, path: str, content: bytes):
+        # Custom logic for writing bytes to a path  using "foo" protocol
+        ...
 
-Then to load a `test.foo` file you can simply use `Cabinet.read`. 
+    @classmethod
+    def _delete_content(cls, path):
+        # Custom logic for deleting the object at a path  using "foo" protocol
+        ...
+
+```
 
-> **NOTE**: In order for the extension to be registered, the class definition must be
-> run at least once. Make sure the modules where your custom `Parser` classes are defined
-> are imported somewhere before they are used.
+Here we define a `FooCabinet`, and register it to the protocol identifier `foo`. Once
+this class is loaded, any `cabinets` function calls using the `foo://` prefix will be
+processed with this class. This means if we called:
 
 ```python
 import cabinets
+from ... import FooCabinet  # ensure FooCabinet is loaded
 
-# .foo file in local filesystem
-local_foo_data = cabinets.read('file://test.foo')
+cabinets.read('foo://example.json')
+```
 
-# .foo file in S3
-s3_foo_data = cabinets.read('s3://test.foo')
+The first call that occurs will be `FooCabinet._read_content('foo.json)`, and that
+result is then parsed by the `JSONParser` before being returned.
+
+> **NOTE**: In order for the protocols to be registered, the class definition must be
+> run at least once. Make sure the modules where your custom `Cabinet` classes are defined
+> are imported somewhere before they are used, OR use the built in [Plugin](#plugins) system.
+
+### Adding Parsers
+
+`cabinets` also supports custom extension parsing in the exact same way:
+
+```python
+from cabinets.parser import Parser, register_extensions
+
+
+@register_extensions('bar')
+class BarParser(Parser):
+    @classmethod
+    def load_content(cls, content: bytes):
+        # Parse bytes from "bar" file format into a Python object
+        ...
+
+    @classmethod
+    def dump_content(cls, data: Any):
+        # Dump a Python object into bytes in the "bar" file format
+        ...
 ```
 
-## Protocol Configuration
+Now if we redo our above example using the `.bar` extension:
+
+```python
+from ... import FooCabinet, BarParser  # ensure FooCabinet and BarParser are loaded
 
-Some storage platform protocols may require some configuration parameters to be set before they can be used.
-Each `Cabinet` subclass can expose a `set_configuration(**config)` classmethod to take 
-care of any required initial setup.
+cabinets.read('foo://example.bar')
+```
+
+This statement is roughly equivalent to:
 
 ```python
-from cabinets.cabinet.s3 import S3Cabinet
+BarParser.load_content(FooCabinet._read_content('foo.bar'))
```

Comment:
I don't think `_read_content()` is supposed to protected here
```suggestion
BarParser.load_content(FooCabinet.read_content('foo.bar'))
```

Reply:
Right now the `Cabinet` class methods are protected. They probably shouldn't be.

Reply:
Now this is right.